android/canonical/app/src/main/java/com/google/samples/quickstart/canonical/MeFragment.kt

```diff
@@ -37,6 +43,33 @@ class MeFragment : Fragment() {
         return inflater.inflate(R.layout.fragment_me, container, false)
     }
 
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val logoutButton : Button = view.findViewById(R.id.logout_button)
+        logoutButton.setOnClickListener {
+            // Sign out both Google account and Firebase
+            FirebaseAuth.getInstance().signOut()
+            (activity as MainActivity).googleSignInClient.signOut().addOnCompleteListener {
+                val intent = Intent(context, MainActivity::class.java)
+                startActivity(intent)
```

Comment:
If this logic were in the ViewModel, it could simplify this code and callbacks to observing sign-in state in the main activity.

Reply:
+1